Kickstarter-iOS/Library/SharedFunctionsTests.swift

```diff
@@ -0,0 +1,44 @@
+import XCTest
+@testable import Kickstarter_Framework
+@testable import Library
+
+internal final class SharedFunctionsTests: XCTestCase {
+  func testLogoutAndDismiss() {
+    let mockAppEnvironment = MockAppEnvironment.self
+    let mockPushNotificationDialog = MockPushNotificationDialog.self
+    let mockViewController = MockViewController()
+
+    XCTAssertFalse(mockAppEnvironment.logoutWasCalled)
+    XCTAssertFalse(mockPushNotificationDialog.resetAllContextsWasCalled)
+    XCTAssertFalse(mockViewController.dismissAnimatedWasCalled)
+    logoutAndDismiss(viewController: mockViewController, appEnvironment: mockAppEnvironment,
+                     pushNotificationDialog: mockPushNotificationDialog)
+    XCTAssertTrue(mockAppEnvironment.logoutWasCalled)
+    XCTAssertTrue(mockPushNotificationDialog.resetAllContextsWasCalled)
+    XCTAssertTrue(mockViewController.dismissAnimatedWasCalled)
+  }
+}
+
+struct MockAppEnvironment: AppEnvironmentType {
```

Comment:
These mock objects should probably be `private` since we don't need to reuse them anywhere else in our test suite.